### PR TITLE
2/3 feature(auto_tuning) Update stragtegy generator for paral_config. 

### PR DESCRIPTION
**Pull request by Antlera**
### What changes were proposed in this pull request?
Update stragtegy generator for paral_config.

### Why are the changes needed?
This is one component of the auto_tuning feature.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
UT.

### CODE REVIEW

**Review by merlintang — COMMENTED** [review]
leave comments

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        gpu_stats=[],
+        model_config=mock_model_config,
+    ) -> ParallelConfig:
+        node_samples = self._extract_node_resource()
+        paral_configs: Dict[str, ParallelConfig] = {}
+        for nodes in node_samples[NodeType.WORKER]:
+            for node in nodes:
+                gpu_stats = node.used_resource.gpu_stats
+                paral_config = node.paral_config
+                data_loader_config, optimizer_config = (
```
Comment (workingloong) [inline dlrover/python/master/hyperparams/simple_strategy_generator.py:87]: Split the line into 2 lines.

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        min_remain_memory = min(
+            entry["total_memory_gb"] - entry["used_memory_gb"]
+            for entry in gpu_stats
+        )
+
+        # Update dataloader configuration version
+        updated_version = dataloader_config.version + 1
+        # Extract dataloader configuration values
+        batch_size = dataloader_config.batch_size
+        logger.info("batch size: %d", batch_size)
```
Comment (workingloong) [inline dlrover/python/master/hyperparams/simple_strategy_generator.py:114]: We can `logger.info(f"batch size: {batch_size}")`

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+        # Extract model configuration values
+        block_size = model_config["block_size"]
+        n_layer = model_config["n_layer"]
+        n_heads = model_config["n_heads"]
+        n_embd = model_config["n_embd"]
+
+        # Calculate the memory required for intermediate activation
+        activation_memory_gb = (
```
[the remark this answers is not in the document]
Comment (Antlera) [reply dlrover/python/master/hyperparams/simple_strategy_generator.py:123]: These magic numbers are from this [blog](https://zhuanlan.zhihu.com/p/624740065). Great analysis for transformer structure model vmem comsumption.

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        )
+
+    def _generate_optimizer_config(self):
+        return OptimizerConfig(5, 6)
+
+    def _extract_node_resource(self) -> Dict[str, List[List[Node]]]:
+        stats = self._stats_collector.get_runtime_stats()
+        node_used_resources: Dict[str, List[List[Node]]] = {}
+        node_used_resources[NodeType.WORKER] = []
+        simple_node = Node(node_type="worker", node_id=0)
```
Comment (workingloong) [inline dlrover/python/master/hyperparams/simple_strategy_generator.py:155]: Is it better to select the node with the max used cuda memory?

`Node(node_type="worker", node_id=0)` ->`Node(node_type=NodeType.WORKER, node_id=0)` 

```
from dlrover.python.common.constants import NodeType
```

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+    def __init__(self, job_uuid):
+        self._job_uuid = job_uuid
+        self._stats_collector = LocalStatsReporter(JobMeta(job_uuid))
+
+    def start(self):
+        try:
+            logger.info("Start local strategy generator")
+            threading.Thread(
+                target=self._extract_node_resource,
```
Comment (workingloong) [inline dlrover/python/master/hyperparams/simple_strategy_generator.py:69]: Why do we need to start a thread to execute `_extract_node_resource`?

Comment (Antlera) [reply dlrover/python/master/hyperparams/simple_strategy_generator.py:69]: Not necessary. This is a leftover during development time.

File: dlrover/python/master/hyperparams/simple_strategy_generator.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        "used_memory_gb": 2.631,
+    },
+]
+
+# Define model configuration as a dictionary
+mock_model_config = {
+    "block_size": 128,
+    "n_layer": 6,
+    "n_heads": 6,
+    "n_embd": 384,
```
[the remark this answers is not in the document]
Comment (Antlera) [reply dlrover/python/master/hyperparams/simple_strategy_generator.py:48]: This is a mock model card configuration. We need to replace it with real model card configuration from model config reporter.

File: dlrover/python/master/hyperparams/simple_strategy_generator.py (hunk 1 above)
Comment (Antlera) [reply dlrover/python/master/hyperparams/simple_strategy_generator.py:87]: Fixed by comment.

File: dlrover/python/master/hyperparams/simple_strategy_generator.py (hunk 2 above)
Comment (Antlera) [reply dlrover/python/master/hyperparams/simple_strategy_generator.py:114]: Fixed by comment.

File: dlrover/python/master/hyperparams/simple_strategy_generator.py (hunk 4 above)
Comment (Antlera) [reply dlrover/python/master/hyperparams/simple_strategy_generator.py]: This is a mock node for standalone mode. The detailed node selection method is in TODO list.

Comment (Antlera) [reply dlrover/python/master/hyperparams/simple_strategy_generator.py]: Indeed, it is reasonable to select the node with the max used Cuda memory.

**Review by workingloong — APPROVED** [review]
LGTM

**Pull request merged**